### PR TITLE
[#12649] When adding a comment while submitting responses, show visibility of the comment as well

### LINKED ISSUE #12649: When adding a comment while submitting responses, show visibility of the comment as well
Context: A respondent (e.g., a student) submitting responses chooses to add an optional comment together with a response.

Problem: Some respondents (based on the comments submitted) seem to think that the comment will be seen by the instructors of the course only. However, the current behavior is that comment has the same visibility as the response itself.

Suggestion: When the comment box is expanded, show the visibility description for the comment as well i.e., simply repeat the visibility description  of the response (seen at the first arrow in the screenshot below) above the comment box as well (i.e., at the second arrow).

image

### PR DESCRIPTION
**Pull request by hallovera**
Fixes #12649

**Outline of Solution**

The problem was that some users could get confused by the visibility of the comments sections, believing they were visible to the instructors only. In fact, they could actually be viewed by the same people as the main question. 

To solve this, the visibility information from the main question is added to the comments section as well. It will show/hide with the comments box, and highlights the people who can see your comments.

When comments are showing: 

image

When comments are hidden:

image

### CODE REVIEW

**Comment by weiquu** [comment]
Hi @hallovera, do fix the lint issues as well as the failing snapshot tests. You can update the snapshots by running` npm run test` and pressing `a` to run all test cases. After that, check through the snapshots to make sure the changes are as expected, then press `u` to update them. You can find more details on snapshot testing [here](https://teammates.github.io/teammates/snapshot-testing.html).

**Comment by jasonqiu212** [comment]
Hi @hallovera, The snapshot tests are still failing. Are you able to fix them following the steps mentioned by @weiquu? Do let us know if you face any troubles!

**Comment by hallovera** [comment]
Sorry for the delay, I believe the tests all pass successfully now. The problem was pretty silly: I misunderstood how snapshot testing worked, and assumed I had to manually fix the issues arising on the tests (which didn't go well). Once I figured out I simply needed to "update" the snapshots, it all fell into place.

Please let me know if there are any other issues! Thanks @weiquu and @jasonqiu212 for your helpful comments.

**Review by weiquu — APPROVED** [review]
LGTM

File: src/web/app/components/question-submission-form/question-submission-form.component.html [hunk 1]
```diff
@@ -146,6 +146,20 @@ <h2 class="question-details"><b>Question {{ model.questionNumber }}: </b>{{ mode
           </div>
 
           <div id="comment-section-qn-{{ model.questionNumber }}-idx-{{ i }}" class="col-12 margin-bottom-20px margin-top-10px indent">
+            <div *ngIf="recipientSubmissionFormModel.commentByGiver">
+              <div class="card-body visibility-card">
+                <p class="text-secondary">The following persons can see your comments:</p>
```
Comment (domlimm) [inline src/web/app/components/question-submission-form/question-submission-form.component.html:151]: Minor nit. Let's standardise with the response block above.
```suggestion
                <p class="text-secondary">Only the following persons can see your comments: </p>
```

**Review by domlimm — COMMENTED** [review]
Small nit. Nice work!

**Comment by domlimm** [comment]
@hallovera Sorry, will need you to update the snapshots please. Thank you!

**Comment by hallovera** [comment]
@domlimm Done!

**Review by domlimm — APPROVED** [review]
LGTM, thank you for your contribution!

**Pull request merged**